### DISCUSSION TITLE
Running NocoDB as an App on Digital Ocean

### DISCUSSION THREAD

**Discussion by KR4STLL** [nocodb/nocodb#3851] [Q&A]
Hi Guys,

I'm very new to self hosting apps, I have set up NocoDB as an App on Digital Ocean by following the guidance here: https://github.com/nocodb/nocodb/discussions/1553

I'm having issues with the app consistently logging me out and giving me white screens whenever I try to really get any further than create or open a project.
Am I missing something or have I done something incorrectly?

Thank you!

**Comment by o1lab** [comment]
Have you set up NC_DB env variable ?

**Reply by KR4STLL** [reply]
I haven't done any other set up, I assumed based on the comment on that discussion that it wasn't necessary

**Reply by KR4STLL** [reply]
"please check out the architecture
https://docs.nocodb.com/#architecture---simple-overview
if NC_DB is not present, we'll create a sqlite for storing the metadata."

**Reply by o1lab** [reply]
So digital ocean Apps feature make those app ephermal. That means they spin down your server when not in use & spin up when a usage starts. And on fresh start, the SQLite database wont be there - hence the logging out. So you would need a database.

Better to use digital ocean droplet. Connect to that terminal. And run our [docker compose](https://docs.nocodb.com/getting-started/installation#docker-compose).

These are the steps
- Create a droplet & connect to the terminal
```
sudo apt install docker.io
sudo apt install docker-compose
wget https://raw.githubusercontent.com/nocodb/nocodb/develop/docker-compose/mysql/docker-compose.yml
docker-compose up -d
```

Visit http://ServerIPAddress:8080

**Reply by jellederijke** [reply]
I run Nocodb on DO's App platform succesfully and I can upgrade without losing data, by correctly setting NC_DB as app env variable, as @o1lab suggested. Highly recommended, since you do not need to manage a droplet's updates etc!
